app/controllers/users_controller.rb

```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+  def destroy
+  	User.find(params['id']).destroy
```

Comment:
Devise has a "Delete Account" link. Why dont you use it?
That will also clear sessions, etc.